### PR TITLE
Change page titles on transition

### PR DESCRIPTION
**Pull request by helfi92**
* Added a component HelmetTitle as a helper to change page titles
* Add a page title to every view

### CODE REVIEW

**Comment by eliperelman** [comment]
I think this is pretty awesome, and almost ready. Could I ask that you move the `<HelmetTitle>`s into the `render` methods of the components where there are `<Client>`s? For example:

```diff
  <Clients credentials={credentials}>
      {clients => (
-       <div>
-         <HelmetTitle title={title} />
           <Component />
```

```diff
  render() {
    return (
      <div>
+       <HelmetTitle title={title} />
    );
  }
```

This should save us an extra nesting of elements.

**Comment by helfi92** [comment]
IndexBrowser, IndexedArtifactBrowser both have `<Client>` but they share the same component. They have different title names so I'm not sure if `<HelmetTitle>` should be inside the component. I left those untouched. Let me know if that's not the best way.

ScopeInspector, SecretsManager and TaskRedirect have many return statements in their render method so it's probably not a good idea to have many `<HelmetTitle>` components. I left that untouched. Again, let me know if there's a better way.

**Comment by eliperelman** [comment]
@helfi92 can you rebase? I've made a few rearrangements to splitting up the component and the view into separate files where appropriate.

File: src/components/HelmetTitle/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Helmet, title } from 'react-helmet';
+
+const HelmetTitle = ({ title }) => (
+  <Helmet>
+    <title>{`${title} | TaskCluster`}</title>
```
Comment (eliperelman) [inline src/components/HelmetTitle/index.js]: Let's make `props.title` conditional.

```js
{props.title ? `${title} | TaskCluster` : 'TaskCluster'}
```

**Review by eliperelman — APPROVED** [review]
:+1:

**Comment by eliperelman** [comment]
:shipit:

**Pull request merged**